src/aws_sigv4_utils.erl

```diff
@@ -0,0 +1,58 @@
+-module(aws_sigv4_utils).
+
+-export([ binaries_join/2
+        , format_time_long/1
+        , format_time_short/1
+        , sha256/1
+        , tolower/1
+        , toupper/1
+        , trimspace/1
+        ]).
+
+-spec binaries_join(binary(), [binary()]) -> binary().
+binaries_join(Separator, Binaries) ->
```

Comment:
Why not use `aws_signature_utils:binary_join/2` which does the exact same thing?

Reply:
Well, this one had several reasons:
1. Parameter order. This one is consistent with `lists:join/1`.
2. I'm not sure the one in `aws_signature_utils` avoids reallocating and copying the same fragments over and over again. It would need to be validated with `erlc +bin_opt_info`. This one explicitly avoids that.

Having said that, yes, the implementations should be merged since they do the same thing.